libraries/api-documenter/src/RenderingHelpers.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import {
+  IDocMethod
+} from '@microsoft/api-extractor/lib/IDocItem';
+
+export class RenderingHelpers {
+
+  /**
+   * Used to validate a data structure before converting to JSON or YAML format.  Reports
+   * an error if there are any undefined members, since "undefined" is not supported in JSON.
+   */
+  // tslint:disable-next-line:no-any
+  public static validateNoUndefinedMembers(json: any, jsonPath: string = ''): void {
+    if (!json) {
+      return;
+    }
+    if (typeof json === 'object') {
+      for (const key of Object.keys(json)) {
+        const keyWithPath: string = jsonPath + '/' + key;
+        // tslint:disable-next-line:no-any
+        const value: any = json[key];
+        if (value === undefined) {
+          throw new Error(`The key "${keyWithPath}" is undefined`);
+        }
+        RenderingHelpers.validateNoUndefinedMembers(value, keyWithPath);
+      }
+    }
+
+  }
+
+  /**
+   * Generates a documentation ID based on the provided scope elements.
+   * A documentation ID is a string that  uniquely identifies an object in
+   * the API documentation web site, and is used e.g. for creating internal hyperlinks.
+   */
+  public static getDocId(packageName: string, exportName?: string, memberName?: string): string {
+    let result: string = RenderingHelpers.getUnscopedPackageName(packageName);
+    if (exportName) {
+      result += '.' + exportName;
+      if (memberName === '__constructor') {
+        result += '.' + '-ctor';
```

Comment:
>result += '.' + '-ctor'; [](start = 8, length = 24)

this seems weird

Reply:
why the leading '-' ?

---
In reply to: [135166978](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135166978) [](ancestors = 135166978)

Reply:
It is a docs.microsoft.com convention.  We can remove it for now.

---
In reply to: [135167001](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135167001) [](ancestors = 135167001,135166978)